### PR TITLE
close API: make classes final

### PR DESCRIPTION
**Pull request by dmaicher**
## Subject

I am targeting this branch, because this breaks BC.

This is part of https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/1187

## Changelog

```markdown
### Changed
- Add final modifier to classes marked as `@final`.
```

### CODE REVIEW

File: src/Builder/ShowBuilder.php [hunk 1]
```diff
@@ -20,9 +20,6 @@
 use Sonata\AdminBundle\Builder\ShowBuilderInterface;
 use Sonata\AdminBundle\Guesser\TypeGuesserInterface;
 
-/**
- * @final since sonata-project/doctrine-orm-admin-bundle 3.24
- */
 class ShowBuilder implements ShowBuilderInterface
```
Comment (VincentLanglet) [inline src/Builder/ShowBuilder.php:23]: final ?

File: tests/Builder/DatagridBuilderTest.php [hunk 2]
```diff
@@ -62,8 +62,12 @@ protected function setUp(): void
         );
 
         $this->admin = $this->createMock(AdminInterface::class);
-        $this->modelManager = $this->createMock(ModelManager::class);
-
+        $this->modelManager = $this->getMockBuilder(ModelManagerInterface::class)
+            ->setMethodsExcept([])
+            // NEXT_MAJOR: remove the "hasMetadata" and "getParentMetadataForProperty" methods and just mock the interface normally
```
Comment (VincentLanglet) [inline tests/Builder/DatagridBuilderTest.php:67]: Why the need to add a NEXT_MAJOR if you're already on the master branch ?

File: tests/Guesser/FilterTypeGuesserTest.php [hunk 3]
```diff
@@ -40,7 +41,11 @@ class FilterTypeGuesserTest extends TestCase
     protected function setUp(): void
     {
         $this->guesser = new FilterTypeGuesser();
-        $this->modelManager = $this->createStub(ModelManager::class);
+        $this->modelManager = $this->getMockBuilder(ModelManagerInterface::class)
```
Comment (VincentLanglet) [inline tests/Guesser/FilterTypeGuesserTest.php:44]: Maybe we should create a DoctrineORMAdminBundle/ModelManagerInterface

Comment (dmaicher) [reply tests/Guesser/FilterTypeGuesserTest.php:44]: yeah It looks like it... we are using methods not defined on `ModelManagerInterface`

File: tests/Builder/DatagridBuilderTest.php (hunk 2 above)
Comment (dmaicher) [reply tests/Builder/DatagridBuilderTest.php:67]: its only needed for some tests that are not cleaned up yet (still with `NEXT_MAJOR` comments). I did not want to do it all in one big PR

File: tests/Guesser/FilterTypeGuesserTest.php (hunk 3 above)
Comment (dmaicher) [reply tests/Guesser/FilterTypeGuesserTest.php:44]: Shall I create an issue about it?

Comment (VincentLanglet) [reply tests/Guesser/FilterTypeGuesserTest.php:44]: Or maybe move this method somewhere else ?

File: tests/Builder/DatagridBuilderTest.php (hunk 2 above)
Comment (dmaicher) [reply tests/Builder/DatagridBuilderTest.php:67]: See https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/90c4d7d59d58e5a7199ce6f6f9015cdaefab7623/tests/Builder/DatagridBuilderTest.php#L136

File: tests/Guesser/FilterTypeGuesserTest.php (hunk 3 above)
Comment (VincentLanglet) [reply tests/Guesser/FilterTypeGuesserTest.php:44]: > Shall I create an issue about it?

Yes

Comment (dmaicher) [reply tests/Guesser/FilterTypeGuesserTest.php:43]: See https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/1296

**Review by OskarStark — DISMISSED** [review]

File: tests/Builder/DatagridBuilderTest.php (hunk 2 above)
Comment (dmaicher) [reply tests/Builder/DatagridBuilderTest.php]: @VincentLanglet removed since you did the cleanup on master already :+1:

**Review by VincentLanglet — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Pull request merged**